### ISSUE TITLE
Support markdown multi-line highlight for constructs with inline-content

### ISSUE THREAD

**Issue by ubitux** [vimwiki/vimwiki#847]
According to latest CommonMark spec as of `0.29` (https://spec.commonmark.org/0.29/#example-635), the following example should be highlighted:

```markdown
*hello
world*
```

With Vimwiki, the `conceal` is not effective here, which hints me that the highlight is not set (would be nice to have a different color for this markup by default).

Similarly, the following doesn't work (which is the one I use since it displays in a different color, even though the semantic is most of the time not correct):

```markdown
`hello
world`
```

Note that the spec says "Line breaks do not occur inside code spans", but this is only referring to the multi-line rendering: if you look at the output, it will still consider the whole thing as inline-code (ignoring the line break).

I use the `gq` command a lot, and thus the markup regularly breaks because of auto inserted line-breaks within highlighted sentences / word-groups.

**Note**: this issue is likely a good candidate for https://github.com/vimwiki/vimwiki/issues/664.

`:VimwikiShowVersion`:

```
Branch: dev
Revision: 78a41b7
Date: 2020-04-03 13:31:36 -0600
```

The issue is also reproducible with `master`.

**Comment by tinmarino** [comment]
Feel free to make a PR. You can get it done in 2h.

**Comment by chipsenkbeil** [comment]
@tinmarino, is this just syntax highlighting for these or is there extra logic to do?

If it's just syntax highlighting, then we could pop in a `syntax region ...` for each of these cases. Is the markdown supported by vimwiki following the commonmark spec? I don't actually know which flavor of markdown is used.

**Comment by ubitux** [comment]
Speaking of this multi-line highlight not supported, there is a similar issue when using `let g:vimwiki_hl_cb_checked = 1`; if you have a multi line entry, only the first line gets highlighted when it's checked:

```markdown
- [x] hello
      world
```

(multiline is actually kind of supported, because when under "world", `o` will add a list entry, and `ctrl`-`space` works to toggle the entry as well)

I can open another issue if needed.

**Comment by tinmarino** [comment]
@chipsenkbeil 
> @tinmarino, is this just syntax highlighting for these or is there extra logic to do?

According to the issue creator, it is just syntax highliting, for me too. But this may break the html conversion that is done line by line.
I never looked at html conversion, but a multiple line stuff is what you've just done that is why I called for your power.

@ubitux 
> I can open another issue if needed.

Yes it would be better: Small issues are easier to close. So it divides better the dev workflow. Please:

1. Link to this issue
2. Explain very well the issue, so that the dev can implement non-regression tests from the description

**Comment by tinmarino** [comment]
@ubitux before opening a new issue, check if [that answer](https://github.com/vimwiki/vimwiki/issues/605#issuecomment-460396710) solves it (found searching `multiline` in issues and PR):

> Try `let g:vimwiki_hl_cb_checked=2`. Admittedly, the doc is a bit inaccurate here.

**Comment by tinmarino** [comment]
@chipsenkbeil yes it would be better as a region as tim pope is doing in the [default markdown syntax](https://github.com/tpope/vim-markdown/blob/9ff8d52c78fb8a587224d7f4d877fc95c8e1bd5b/syntax/markdown.vim#L100-L105) : it is much cleaerer

**Comment by tinmarino** [comment]
Fixed:  (1b16720 or search "Typeface" in commit title or Last commit refering this issue)

The syntax are redefined as regions which permited to solve this (see $VIMRUNTIME/wiki/markdown.vim from tim Pope)
This permited to solve this issue and #847 (receiving the same comment)

**Comment by tinmarino** [comment]
Ok this one is not finished:
TODO:
```
`hello
world`

**Comment by tinmarino** [comment]
Closed in favor of #709 for the "inline" code than spam multiple line (for easyer tracking)

Thanks @ubitux for your very descriptive issues.
